.lagoon.yml

```diff
@@ -16,68 +16,60 @@ environments:
 
 tasks:
   # Pre-rollout tasks do not get called on first deploy; use post-rollout tasks instead.
+  # Pre-rollout tasks need to have file existance checks as these are run in the active
+  # containers and it is not guaranteed that the commands will be available there.
   pre-rollout:
-    - run:
-        name: Ensure backup folder exists - not all current envs have one
-        command: mkdir -p /app/web/sites/default/files/private/backups
+    -
+      run:
+        name: Pre-rollout database updates
+        command: "[ -f /app/vendor/bin/govcms-pre-deploy-db-update ] && /app/vendor/bin/govcms-pre-deploy-db-update || echo 'Pre Update databse is not available.'"
         service: cli
-        shell: bash    
-    - run:
+        shell: bash
+    -
+      run:
         name: Snapshot the database and store
-        command: if [[ "$LAGOON_ENVIRONMENT_TYPE" = "production" ]]; then drush sql:dump --root=/app --gzip --result-file=/app/web/sites/default/files/private/backups/pre-deploy-dump.sql; fi
+        command: "[ -f /app/vendor/bin/govcms-db-backup ] && /app/vendor/bin/govcms-db-backup || echo 'Database backup is not available.'"
```

Comment:
Does this mean that the first time we update the scaffold, there will be no pre-deploy backup to rollback to - but obviously on subsequent deploys there will be

Reply:
Yeah - unfortunately that scaffold tooling project will not be included in the running containers and because the pre-rollout is executed in the current container we can't be certain that the file will exist.

We _could_ have an else to fallback to the standard `drush sql:dump` to deprecate and refactor this later though.

Reply:
Although, with the staggered approach of this rollout - if we're not including this in the scaffolds with 8.5.0. Scaffold tooling will be available when we merge this so it should be completely okay to run the backup this way.